### ISSUE TITLE
Profiler setup for virtual machines is a Rube Goldberg machine, doesn't work in many scenarios

### ISSUE THREAD

**Issue by peter-bertok** [microsoft/ApplicationInsights-dotnet#2193]
I've spent the last two days trying to convince Application Insights to collect profiler traces from an Azure virtual machine, a scenario that "should just work".

I'm following the instructions in the following article: https://docs.microsoft.com/en-us/azure/azure-monitor/app/profiler-vm

Some notes:

- There's a requirement to install an unrelated monitoring agent. Why?
- Then there's a step for **manually** editing that agent's config file. What about ARM template deployments?
- The documentation helpfully provides a JSON sample for what is an XML config file in the (now outdated) ARM template samples.
- The provided snippet in the documentation is out-of-context. "Just put it in there... somewhere."
- Apparently the SDK integration is also required. What about codeless monitoring!?
- App Insights supports an instrumentationKeyMap setting, so that multiple virtual apps can send telemetry to different instances of App Insights. However, the WADCfg has only a single sink for "ApplicationInsightsProfiler"... so... this can't ever work for multi-app servers.

To summarize, after days of effort, I could not enable a basic feature that _should just work_ out of the box, just like it does with New Relic. The process is completely impractical within ARM templates, and will never work in quite ordinary scenarios.

Am I missing something?

**Comment by cijothomas** [comment]
If you're seeing issue with [Application Insights Profiler](https://docs.microsoft.com/azure/azure-monitor/app/profiler-overview), get support from [Azure Support Center](https://azure.microsoft.com/support/create-ticket/). This repo does not host profiler code, and is only for SDK issues.

**Comment by segannon** [comment]
Hi @peter-bertok, if you'd like to reach out to the Service Profiler team directly for questions or issues you can use the email serviceprofilerhelp@microsoft.com. I'll review the documentation based on your suggestions and try to make some clarifying updates. Thanks for taking the time to post.
